src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -40,4 +43,25 @@ public OverseasEntityDataApi getOverseasEntityData(String requestId, String comp
             throw new ServiceException(e.getMessage(), e);
         }
     }
+
+    public CompanyProfileApi getCompanyProfile(String requestId, String companyNumber)
+            throws ServiceException {
+        var dataMap = new DataMap.Builder().companyNumber(companyNumber).build();
+        try {
+            var companyProfileApi = apiClientService
```

Comment:
the getCompanyProfile method shouldn't be in the PrivateDataRetrievalService anymore so we'll need a PublicDataRetrievalService to house it.

Reply:
Separated the code.